src/pytorch_lightning/utilities/migration/utils.py

```diff
@@ -111,7 +112,10 @@ def _pl_migrate_checkpoint(checkpoint: _CHECKPOINT, checkpoint_path: Optional[_P
 
     # include the full upgrade command, including the path to the loaded file in the error message,
     # so user can copy-paste and run if they want
-    path_hint = os.path.relpath(checkpoint_path, os.getcwd())
+    if not _IS_WINDOWS:  # side-step bug: ValueError: path is on mount 'C:', start on mount 'D:'
```

Comment:
what about `Path.cwd().relative_to(checkpoint_path)`, does it have the same issue?

Reply:
Yes:

```python
p = Path('C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_logger_reset_correctly_LR0\\.lr_find_c91f5435-ea1d-4fe7-8ad8-4229daec1a4d.ckpt')
cwd = Path('D:\\a\\lightning\\lightning\\tests\\tests_pytorch')
cwd.relative_to(p)
Traceback (most recent call last):
  File "/home/carmocca/git/py310/lib/python3.10/site-packages/IPython/core/interactiveshell.py", line 3433, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-7-f6f1c7cc80c1>", line 1, in <module>
    cwd.relative_to(p)
  File "/usr/lib/python3.10/pathlib.py", line 818, in relative_to
    raise ValueError("{!r} is not in the subpath of {!r}"
ValueError: 'D:\\a\\lightning\\lightning\\tests\\tests_pytorch' is not in the subpath of 'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_logger_reset_correctly_LR0\\.lr_find_c91f5435-ea1d-4fe7-8ad8-4229daec1a4d.ckpt' OR one path is relative and the other is absolute.
```

Reply:
Oh but if checkpoint_path is an absolute path with a different stem from cwd, then computing the relative path should not be possible for `os.path.relpath`. This is an edge case I didn't consider when writing this function. 

The motivation was to shorten path not to make it too long, so relative to the cwd. But if the checkpoint path and cwd don't have a common stem, then that doesn't make sense. 

We can fix this by checking

```py
if checkpoint_path.is_relative_to(Path.cwd)
    ...
```

Reply:
Is this what you were thinking? [`eb51fe9` (#16164)](https://github.com/Lightning-AI/lightning/pull/16164/commits/eb51fe9fae3db542b8c784aff35a4f9d3b50981a)

Reply:
NVM. `is_relative_to` was introduced in 3.9 so we cannot use it.

Reply:
Yes exactly, that was the original intention. No need to block the PR for the hotfix, but we might need to revisit.